### DISCUSSION TITLE
Zluda performance

### DISCUSSION THREAD

**Discussion by Omen090909** [lshqqytiger/stable-diffusion-webui-directml#408] [General]
Hi, i managed to use zluda but performance is basically exactly same as i had with directml
Am using rx 6700, and have the same performance also in sd next
What are your experiences with zluda guys?
The only upside for me is better memory management, but it still seems limited compared to what happenes on nvidia gpus

**Comment by Eminic** [comment]
Rx 6700 XT here, with zluda i get around 3.7 seconds/it with an SDXL model. Compared to rocm on linux, it's about 3X slower. Same performance in SD Next. Personally sticking with linux until it approaches anywhere near that performance.

**Reply by Omen090909** [reply]
thanks for info, seems around same for me
will try linux soon, but i ve heard it can be limited in some features, or is it fine now?
btw what image size give you that performance? i get similar with 768x1024 on dpm++ 2m karras, but twice as slow on dpm++ sde karras, but it gives really awesome quality in sdxl

**Reply by Eminic** [reply]
Linux isn't as simple to use especially if you're already familiar with windows, which is why i dualboot it (windows on one drive linux on the other drive only for stable diffusion). It takes a bit of learning but afterwards it's quite straightforward. 

I've also got guides on how to install rocm drivers and stable diffusion on it if you're interested. (https://www.youtube.com/watch?v=NKR_1TUO6go)

I've found that running an SDXL model with an image size of 832x1216 (the total needs to add up to 1024+1024 for SDXL models), on euler a I get around 1.6/1.7 seconds per iteration with stable diffusion forge, and 1.2/1.3 seconds per iteration with the normal a1111. 

Quite a significant difference in performance compared to windows which is why i won't switch over just yet.

**Reply by Omen090909** [reply]
ye, i thought i d get away with vm for linux, but that won't be that simple in that case ;d
i watch that guy on yt so will probably look into that when i get to it
i can't find my pendrive, borrowed another one and it is broken xD so unless i find another way to install it i d need to wait until i find it or buy a new one.
Thanks very much for iteration times, would have to have them, also bit sad i didn't get any performance boost with zluda here.
What about other features in auto1111, i ve heard some didn't work on linux, did you happen to run into some problems? let's say with controlnet or some upscaling/inpainting?

**Reply by Eminic** [reply]
I would say it's quite the opposite, on linux you get ALL the features of a1111. Forge for example is a modified version of a1111 with even more features. The only thing you should be aware of is memory issues, especially if you plan on using SDXL. I've seen that on the normal a1111 my card often uses 10+ gb VRAM, and my ram is also quite high. Forge helps a ton with that but you do lose a bit of performance in exchange, worth it in my opinion though.

**Reply by Omen090909** [reply]
thanks, that's good to hear

**Reply by Soulreaver90** [reply]
I bought a spare SSD for $50 and installed Linux on it (Ubuntu 22.04). Been getting the full power of my 6700xt and native support of web-ui. Honestly it isn’t that hard and it’s worth it since most of these developers focus on Linux first.

**Reply by Omen090909** [reply]
i had a spare ssd for it and tried it, and well, the lack of customisation for gpu kinda killed my interest, and i can't run it stock cause it's gonna burn. I might still try it, maybe find a workable sollution, but so far been quite disappointed with how linux works

**Reply by Eminic** [reply]
There's an program called core ctrl for linux which is very similar to msi afterburner in terms of functionality, controls frequencies, power delivery and you can make custom fan curves as well. I've been using it myself since my gpu also melts on stock settings.

**Reply by Omen090909** [reply]
thanks, i ll try it, altho i m sceptical about outside programs to change voltage on gpu, and i don't think i can find tons of footage on yt like i can for everyting on windows :x

**Comment by pinea00** [comment]


**Reply by Omen090909** [reply]
idk, i tried deep-cache in sd next, speed might be doubled, but quality is halfed
also yes, sd next is super unstable for me too, can't run sdxl at all (i mean i m getting pixels instead of image)

**Comment by Bercraft** [comment]
I do not know if its the good place but....... here [https://github.com/brknsoul/ROCmLibs](url) there are rocm libraries builds.
Ipersonally own a w5700 pro and it works wonders.

**Comment by pw405** [comment]
First test of Zluda on 7900XTX and uhh.. **WOW, blown away!!**  

With SDXL model AeitherverseXL (https://civitai.com/models/308337/aetherverse-xl) 
Generated a base image at 1600 X 768. Params: Steps: 30, Sampler: Restart, Schedule type: Exponential, CFG scale: 7 

This was taking ~ 3-4 minutes on DirectML.  Just finished TWO images for a **total 54 seconds**. 

Next, tested Ultimate SD Upscale to increase to size 3X to 4800 X 2304.  It only took 1 minute & 49 seconds for 18 tiles, 30 steps each! WOW! This could easily take ~8+ minutes or more on DirectML. 

So far, ZLUDA looking to be a game changer.  GPU runs cooler, it doesn't bog the entire system down and lock up before finishing every image.  VERY excited to see where this goes!

**Reply by pinea00** [reply]
To use forge Zluda and use DPM karrs turbo sampler,you will wow again.

**Reply by pw405** [reply]
Is that sampler just not available in A1111?  What sort of it/second do you get with it?

**Reply by pinea00** [reply]
RX6600
Generating 1024*1024 SDXL A1111-zluda (FP8) takes more than 2 minutes
Generating 1024*1024 SDXL forge-zluda takes 1 minute and 27 seconds
Generating 1024*1024 SDXL forge-zluda (DPM++ 2M SDE Turbo) takes about 43s

**Reply by lshqqytiger** [reply]
ROCm 5.7 does not support FP8 datatype and operations.

**Comment by gigend** [comment]
support Radeon RX 570 vram 4gb? how performance, better than directml?

**Reply by pw405** [reply]
I haven't heard of anybody running Zluda on an R570 - but its worth a shot to try, it can help VRAM use. 

For reference - if I'm doing big image with SDXL - my VRAM allocation would always be maxed out with DirectML, but with ZLUDA I only see about ~12 GB of VRAM allocation. 

Also be sure to use Scaled Dot Product in the optimizations settings within the Web UI.

**Comment by patientx** [comment]
> ROCm 5.7 does not support FP8 datatype and operations.

Just came back to your app once again and trying zluda now, it automatically using doggettx , is that correct for zluda and do we need any specific options or commandline parameters other than zluda for an 8gb 6600 , maybe lowvram etc ? Thanks.

**Reply by lshqqytiger** [reply]
use scaled-dot product, and you don't need lowvram unless you are trying large images
